go/mysqlconn/client_test.go

```diff
@@ -194,7 +194,11 @@ func TestWithRealDatabase(t *testing.T) {
 		},
 	}
 	if !reflect.DeepEqual(result, expectedResult) {
-		t.Errorf("unexpected result for select, got:\n%v\nexpected:\n%v\n", result, expectedResult)
+		// MySQL 5.7 is adding the NO_DEFAULT_VALUE_FLAG to Flags.
+		expectedResult.Fields[0].Flags |= 4096
```

Comment:
expectedResult is only use for this comparison here, not later on by other flavors.